client/deposit_ethereum.go

```diff
@@ -0,0 +1,467 @@
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	var vaultResponse VaultResponse
+	err = json.NewDecoder(resp.Body).Decode(&vaultResponse)
+
+	if vaultResponse.Status == ERROR {
+		if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+			return 0, &ErrClient{
+				Status: resp.StatusCode,
+				Err:    fmt.Errorf(vaultResponse.Message),
+			}
+		} else if resp.StatusCode >= 500 {
+			return 0, &ErrServer{
+				Status: resp.StatusCode,
+				Err:    fmt.Errorf(vaultResponse.Message),
+			}
+		}
+
+		return 0, fmt.Errorf("status: %d\nerror: %s", resp.StatusCode, vaultResponse.Message)
+
+	} else if err != nil {
+		return 0, &ErrJSONDecoding{Err: err}
+	}
+
+	return vaultResponse.Payload.ID, nil
+}
+
+type CryptoDepositRequest struct {
+	Amount                 string `json:"amount"`
+	StarkAssetID           string `json:"token_id"`
+	StarkPublicKey         string `json:"stark_key"`
+	DepositBlockchainHash  string `json:"deposit_blockchain_hash"`
+	DepositBlockchainNonce string `json:"deposit_blockchain_nonce"`
+	VaultID                int    `json:"vault_id"`
+}
+
+type CryptoDepositResponse struct {
+	Status  Status `json:"status"`
+	Message string `json:"message"`
+	Payload interface{} `json:"payload"`
+}
+
+// CryptoDepositStart
+func (c *Client) cryptoDepositStart(ctx context.Context, depositReq CryptoDepositRequest) (CryptoDepositResponse, error) {
+	err := c.CheckAuth()
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	requestBody, err := json.Marshal(depositReq)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.cryptoDepositStartURL.String(), bytes.NewReader(requestBody))
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("JWT %s", c.jwtToken))
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+	defer resp.Body.Close()
+
+	var cryptoDepositResponse CryptoDepositResponse
+	err = json.NewDecoder(resp.Body).Decode(&cryptoDepositResponse)
+
+	if cryptoDepositResponse.Status == ERROR {
+		if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+			return CryptoDepositResponse{}, &ErrClient{
+				Status: resp.StatusCode,
+				Err:    fmt.Errorf(cryptoDepositResponse.Message),
+			}
+		} else if resp.StatusCode >= 500 {
+			return CryptoDepositResponse{}, &ErrServer{
+				Status: resp.StatusCode,
+				Err:    fmt.Errorf(cryptoDepositResponse.Message),
+			}
+		}
+
+		return CryptoDepositResponse{}, fmt.Errorf("status: %d\nerror: %s", resp.StatusCode, cryptoDepositResponse.Message)
+
+	} else if err != nil {
+		return CryptoDepositResponse{}, &ErrJSONDecoding{Err: err}
+	}
+
+	return cryptoDepositResponse, nil
+}
+
+func (c *Client) SetAllowance(amount int) {
```

Comment:
is this used anywhere?

Reply:
This will be used the user to set the allowance

Reply:
Anyways will be changing it